tests/unit/test_gitlab_http_methods.py

```diff
@@ -372,6 +372,63 @@ def response_callback(
     assert "http://example.com/api/v4/user/status" in error_message
 
 
+def test_http_request_on_409_resource_lock_retries(gl_retry):
+    url = "http://localhost/api/v4/user"
+    retried = False
+
+    def response_callback(
+        response: requests.models.Response,
+    ) -> requests.models.Response:
+        """We need a callback that adds a resource lock reason only on first call"""
+        nonlocal retried
+
+        if not retried:
+            response.reason = "Resource lock"
+
+        retried = True
+        return response
```

Comment:
We're mostly doing this because `responses` doesn't support mocking `Response.reason` as @pspacek already found out in https://github.com/python-gitlab/python-gitlab/issues/2325#issuecomment-1282343954.